alibuild_helpers/build.py

```diff
@@ -792,51 +791,60 @@ def downloadTask(p):
     if spec["package"] == mainPackage:
       mainBuildFamily = spec["build_family"]
 
+    busyRevisions = set()
+    # We can tell that the remote store is read-only if it has an empty or
+    # no writeStore property. See below for explanation of why we need this.
+    revisionPrefix = "" if getattr(syncHelper, "writeStore", "") else "local"
     for d in packages:
-      realPath = readlink(d)
-      matcher = format("../../%(a)s/store/[0-9a-f]{2}/([0-9a-f]*)/%(p)s-%(v)s-([0-9]*).%(a)s.tar.gz$",
-                       a=args.architecture,
-                       p=spec["package"],
-                       v=spec["version"])
-      m = re.match(matcher, realPath)
-      if not m:
+      match = re.match(
+        "../../%s/store/[0-9a-f]{2}/([0-9a-f]+)/%s-%s-((?:local)?[0-9]+).%s.tar.gz$" %
+        tuple(map(re.escape, (args.architecture, spec["package"],
```

Comment:
`[re.escape(x) for x in [args.architecture, spec["package"], spec["version"], args.architecture]]` but then again, why not simply using `str.format`?

Reply:
Reverted or now; I'll include it in the separate cleanup PR.